required argument 'flush_errors' error with latest version of mypy

Hi @dbader,

`pytest-mypy` seems to be broken with the latest release of `mypy==0.570`, but I did not yet have the time to investigate it further or fix it. It yields the following errors for each file:

```
[...]: in runtest
    res = mypy.main.type_check_only(sources, None, options)
E   TypeError: type_check_only() missing 1 required positional argument: 'flush_errors'
```

My builds using pytest-mypy are now failing with the same error message:

```.tox/py36/lib/python3.6/site-packages/pluggy/__init__.py:617: in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
.tox/py36/lib/python3.6/site-packages/pluggy/__init__.py:222: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
.tox/py36/lib/python3.6/site-packages/pluggy/__init__.py:216: in <lambda>
    firstresult=hook.spec_opts.get('firstresult'),
.tox/py36/lib/python3.6/site-packages/_pytest/runner.py:106: in pytest_runtest_call
    item.runtest()
.tox/py36/lib/python3.6/site-packages/pytest_mypy.py:58: in runtest
    res = mypy.main.type_check_only(sources, None, options)
E   TypeError: type_check_only() missing 1 required positional argument: 'flush_errors'```

--- a/.travis.yml
+++ b/.travis.yml
@@ -3,7 +3,6 @@
 sudo: false
 language: python
 python:
-  - "3.3"
   - "3.4"
   - "3.5"
   - "3.6"

--- a/pytest_mypy.py
+++ b/pytest_mypy.py
@@ -5,7 +5,7 @@ TODO:
     [ ] Proper docs
 """
 import pytest
-import mypy.main
+import mypy.api
 
 
 def pytest_addoption(parser):
@@ -36,9 +36,7 @@ class MypyItem(pytest.Item, pytest.File):
 
     def runtest(self):
         """
-        Run mypy on the given file. Because mypy's API is in flux
-        we emulate parts of mypy.main.main() here and leave it to
-        mypy to process it's options, for example.
+        Run mypy on the given file.
         """
         # TODO: This should be hidden behind a debug / verbose flag.
         print('Running mypy on', self.path)
@@ -52,16 +50,9 @@ class MypyItem(pytest.Item, pytest.File):
             #       Or maybe we'll just rely on mypy.ini being present?
         ]
 
-        sources, options = mypy.main.process_options(mypy_argv)
-
-        try:
-            res = mypy.main.type_check_only(sources, None, options)
-            errors = res.errors
-        except mypy.errors.CompileError as e:
-            errors = e.messages
-
-        if errors:
-            raise MypyError('\n'.join(errors))
+        stdout, _, _ = mypy.api.run(args=mypy_argv)
+        if stdout:
+            raise MypyError(stdout)
 
     def repr_failure(self, excinfo):
         """

--- a/setup.py
+++ b/setup.py
@@ -31,7 +31,6 @@ setup(
         'Topic :: Software Development :: Testing',
         'Programming Language :: Python',
         'Programming Language :: Python :: 3',
-        'Programming Language :: Python :: 3.3',
         'Programming Language :: Python :: 3.4',
         'Programming Language :: Python :: 3.5',
         'Programming Language :: Python :: 3.6',

--- a/setup.py
+++ b/setup.py
@@ -23,7 +23,7 @@ setup(
     description='Mypy static type checker plugin for Pytest',
     long_description=read('README.rst'),
     py_modules=['pytest_mypy'],
-    install_requires=['pytest>=2.9.2', 'mypy>=0.470'],
+    install_requires=['pytest>=2.9.2', 'mypy~=0.570'],
     classifiers=[
         'Development Status :: 4 - Beta',
         'Framework :: Pytest',

--- a/tox.ini
+++ b/tox.ini
@@ -1,6 +1,6 @@
 # For more information about tox, see https://tox.readthedocs.io/en/latest/
 [tox]
-envlist = py33,py34,py35,flake8
+envlist = py34,py35,py36,flake8
 
 [testenv]
 deps = pytest

